lib/criteo-api-retail.d.ts

```diff
@@ -0,0 +1,172 @@
+declare module 'criteo-api-retail';
+
+interface MetaData {
+    totalItemsAcrossAllPages: number,
+    currentPageSize: number,
+    currentPageIndex: number,
+    totalPages: number,
+    nextPage?: string
+}
+
+interface PageInfo {
+    pageIndex: number,
+    pageSize: number
+}
+
+interface AccountsResponse {
+    data: [
+        {
+            id: string,
+            type: string,
+            attributes: {
+                name: string
+                type: string,
+                subtype: string
+                countries: [
+                    string
+                ],
+                currency: string,
+                parentAccountLabel: string,
+                timeZone: string
+            }
+        }
+    ],
+    metadata: MetaData
+}
+
+interface CampaignAttributes {
+    accountId: string,
+    promotedBrandIds: string[],
+    budgetSpent: number,
+    budgetRemaining: number,
+    status: "active" | "inactive",
+    createdAt: string,
+    updatedAt: string,
+    type?: "auction" | "preferred",
+    drawableBalanceIds: string[],
+    clickAttributionWindow: "7D" | "14D" | "30D",
+    viewAttributionWindow: "none" | "1D" | "7D" | "14D" | "30D",
+    name: string,
+    budget: number | null
+}
+
+type RequireOnly<T, K extends keyof T> = Required<Pick<T, K>> & Partial<Omit<T, K>>;
+
+type CreateCampaign = RequireOnly<CampaignAttributes, "name">;
+
+type UpdateCampaign = RequireOnly<CampaignAttributes, "name" | "budget" | "clickAttributionWindow" | "viewAttributionWindow">
+
+interface CampaignResponse {
+    data: {
+        id: string,
+        type: "RetailMediaCampaign",
+        attributes: CampaignAttributes
+    }
+}
+
+interface CampaignsResponse {
+    data: [{
+        id: string,
+        type: "RetailMediaCampaign",
+        attributes: CampaignAttributes
+    }],
+    metadata: MetaData
+}
```

Comment:
Based on the Response/PagedResponse interface suggestions above...

```suggestion
interface CampaignData {
  id: string,
  type: "RetailMediaCampaign",
  attributes: CampaignAttributes
}

type CampaignResponse = Response<CampaignData>;
type CampaignsResponse = PagedResponse<CampaignData[]>;
```